configure.ac

```diff
@@ -169,16 +169,17 @@ AC_FUNC_REALLOC
 AC_CHECK_FUNCS([inet_ntoa strdup])
 AC_CHECK_FUNCS([strlcpy strlcat setgroups])
 
-dnl Enable extra warnings
-DESIRED_FLAGS="-Wall -Wextra"
+dnl Base flags
+BASE_FLAGS="-Wall -Wextra"
+dnl Enable extra warnings and other things
+EXTRA_FLAGS="-fdiagnostics-show-option -Wno-unused-parameter -Wmissing-prototypes -Wstrict-prototypes -Wmissing-declarations -Wfloat-equal -Wundef -Wformat=2 -Wlogical-op -Wmissing-include-dirs -Wformat-nonliteral -Wold-style-definition -Wpointer-arith -Waggregate-return -Winit-self -Wpacked --std=c89 -ansi -pedantic -Wno-overlength-strings -Wc++-compat -Wno-long-long -Wno-overlength-strings -Wdeclaration-after-statement -Wredundant-decls -Wmissing-noreturn -Wshadow -Wendif-labels -Wcast-qual -Wcast-align -Wwrite-strings -Wp,-D_FORTIFY_SOURCE=2 -fno-common"
 
 if test -n "${MAINTAINER_MODE_FALSE}"; then
-   DESIRED_FLAGS="-Werror $DESIRED_FLAGS -fdiagnostics-show-option -Wno-unused-parameter -Wmissing-prototypes -Wstrict-prototypes -Wmissing-declarations -Wfloat-equal -Wundef -Wformat=2 -Wlogical-op -Wmissing-include-dirs -Wformat-nonliteral -Wold-style-definition -Wpointer-arith -Waggregate-return -Winit-self -Wpacked --std=c89 -ansi -pedantic -Wno-overlength-strings -Wc++-compat -Wno-long-long -Wno-overlength-strings -Wdeclaration-after-statement -Wredundant-decls -Wmissing-noreturn -Wshadow -Wendif-labels -Wcast-quual -Wcast-align -Wwrite-strings -Wp,-D_FORTIFY_SOURCE=2 -fno-common"
+   EXTRA_FLAGS="-Werror $EXTRA_FLAGS"
```

Comment:
That does not really make it better.
You are also mixing up stuff here: Sneaking in changes under a commit message indicating a different change... ;-)

I want us to **always** build with as many of the warning flags as supported.
The **only** difference in maintainer-mode would be the presence of `-Werror`.

I could also not care less about a 2 seconds time for configure. If this would reduce it from 2 minutes to 2 seconds, I'd care. So what's the point?